packages/build-utils/test/unit.get-path-override-for-package-manager.test.ts

```diff
@@ -66,6 +66,39 @@ describe('Test `getPathOverrideForPackageManager()`', () => {
     });
   });
 
+  describe('with package.json#engines.pnpm', () => {
```

Comment:
Can we add a test where `packageJsonEngines: { pnpm: '>=9.0.0' }` is specied, but `corepackPackageManager` is not?